loom/engine/bindings/loom/lmSQLite.cpp

```diff
@@ -0,0 +1,384 @@
+        }
+        lua_pushnumber(L, result);
+
+        return 1;
+    }
+
+    int step()
+    {
+        int result = sqlite3_step(statementHandle); 
+        if((result == SQLITE_ERROR) || (result == SQLITE_MISUSE))
+        {
+            lmLogError(gSQLiteGroup, "Error calling bindString for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;
+    }
+
+    int columnType(int col)
+    {
+        return sqlite3_column_type(statementHandle, col);
+    }
+
+    int columnInt(int col)
+    {
+        return sqlite3_column_int(statementHandle, col);
+    }
+
+    double columnDouble(int col)
+    {
+        return sqlite3_column_double(statementHandle, col);
+    }
+
+    const char* columnString(int col)
+    {
+        const char *text = (const char *)sqlite3_column_text(statementHandle, col);
+        if(text == NULL)
+        {
+            return NULL;
+        }
+
+        //copy the string into our static return array so it survives the trip to Loomscript Land!
+        strcpy(szReturnString, text);
+        return szReturnString;
+    }
+
+    int columnBytes(lua_State *L)
+    {
+        //get the column
+        int col = (int)lua_tonumber(L, 1);
+
+        //get the blob from the column
+        void *blob = (void *)sqlite3_column_blob(statementHandle, col);
+        if(blob == NULL)
+        {
+            lua_pushnil(L);
+            return 1;
+        }
+        int size = sqlite3_column_bytes(statementHandle, col);
+
+        //valid blob so allocate byte array for it
+        utByteArray *bytes = new utByteArray();
+        bytes->allocateAndCopy(blob, size);
+
+        //go lua go!
+        lualoom_pushnative<utByteArray>(L, bytes);
+        return 1;
+    }
+
+    int reset()
+    {
+        int result = sqlite3_reset(statementHandle); 
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling reset for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;        
+    }
+
+    int finalize()
+    {
+        int result = sqlite3_finalize(statementHandle); 
+        if(result != SQLITE_OK)
+        {
+            lmLogError(gSQLiteGroup, "Error calling finalize for database: %s with Result Code: %i", parentDB->getDBName(), result);
+        }
+        return result;        
+    }
+};
+
+char Statement::szReturnString[4096] = "";
+
+
+
+
+//**Connection** external function initialisation
+Statement *Connection::prepare(const char *query)
+{
+    Statement *s = new Statement(this);
+
+    //prepare the database with the query provided
+    int res = sqlite3_prepare_v2(dbHandle, query, -1, &s->statementHandle, NULL);
+    if(res != SQLITE_OK)
+    {
+        lmLogError(gSQLiteGroup, "Error preparing the SQLite database: %s with message: %s", szDBName, getErrorMessage());
+    }
+    return s;
```

Comment:
Not sure what you mean exactly here... The message in theology will be the
text for the last error. Last error is set and valid on Connection calls
that don't return OK.
On 12 Feb 2015 5:34 AM, "Ben Garney" <notifications@github.com> wrote:

> In loom/engine/bindings/loom/lmSQLite.cpp
> <https://github.com/LoomSDK/LoomSDK/pull/261#discussion_r24556275>:
>
> > +
> > +
> > +
> > +
> > +//**Connection** external function initialisation
> > +Statement *Connection::prepare(const char *query)
> > +{
> > +    Statement *s = new Statement(this);
> > +
> > +    //prepare the database with the query provided
> > +    int res = sqlite3_prepare_v2(dbHandle, query, -1, &s->statementHandle, NULL);
> > +    if(res != SQLITE_OK)
> > +    {
> > +        lmLogError(gSQLiteGroup, "Error preparing the SQLite database: %s with message: %s", szDBName, getErrorMessage());
> > +    }
> > +    return s;
>
> Do we need to use the getLastError API here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/LoomSDK/LoomSDK/pull/261/files#r24556275>.
>